cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -12,60 +15,99 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
-    }
+      def extractJfrSettings(event: RecordedEvent): Unit = {
+
+        val id:Long = event.getValue("id")
+        val name:String = event.getValue("name")
+        val value:String = event.getValue("value")
+
+        if( name.equals("enabled") ) {
+          id match {
+
+            case RecordedEventTypeIdMapping.AllocationInNewTLAB =>
+              if(value.toBoolean) AllocationInNewTlabEnabled=Some(true)
+              else throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation in new TLAB' event not enabled")
+
+            case RecordedEventTypeIdMapping.AllocationOutsideTlab =>
+              if(value.toBoolean) AllocationOutsideTlabEnabled=Some(true)
+              else throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation outside TLAB' event not enabled")
+
+            case RecordedEventTypeIdMapping.MethodProfilingSample =>
+              if(value.toBoolean) MethodProfilingSampleEnabled=Some(true)
+              else throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Method Profiling Sample' event not enabled")
 
-    def foundRequiredEvents(): Unit = {
-      if(!allocationInNewTlab){
-        throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation in new TLAB' event not found")
+            case _ =>
+          }
+        }
       }
-      if(!allocationOutsideTlab){
-        throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation outside TLAB' event not found")
+
+      def notDoneFetchingInfo():Boolean = {
+        AllocationInNewTlabEnabled.isEmpty ||  AllocationOutsideTlabEnabled.isEmpty || MethodProfilingSampleEnabled.isEmpty ||
+          operatingSytem.isEmpty || classPaths.size()==0
       }
-      if(!methodProfilingSample){
-        throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Method Profiling Sample' event not found")
+
+      val recordingFile = new RecordingFile(file.toPath)
+      while(recordingFile.hasMoreEvents && notDoneFetchingInfo()){
+        val event = recordingFile.readEvent()
+        event.getEventType.getId match {
+
+          case RecordedEventTypeIdMapping.InitialSystemProperty => extractClassPath(event)
+          case RecordedEventTypeIdMapping.RecordingSetting => extractJfrSettings(event);
+
+          case RecordedEventTypeIdMapping.OSInformation =>
+            operatingSytem = if(event.getValue("osVersion").toString.toLowerCase().contains("win"))  Some("win")
+            else Some("other")
+          case _ =>
+        }
       }
+      recordingFile.close()
     }
 
-    try{
-      foundRequiredEvents()
-    }catch {
-      case e: JfrEventNotFoundException => e.printStackTrace(); System.exit(-1)
+    preProcessing()
+
+    val recordingFile = new RecordingFile(file.toPath)
+    val classLoader = new URLClassLoader(classPaths.stream().distinct().collect(Collectors.toList[URL]).toArray(new Array[URL](0)))
```

Comment:
we want to share the classloader, when the original classpath is the same

Reply:
It will be shared. If the classpaths are the same the ClassLoader's hash value is equal. the will all be using the same hash to get a classLoader object from the map of ClassLoaders

Reply:
yes - but the classpaths could be different and have the same hash